wlroots/wlr_types/texture.py

```diff
@@ -1,6 +1,23 @@
 # Copyright (c) Sean Vig 2020
 
+from wlroots import lib
+from wlroots.renderer import Renderer
+
 
 class Texture:
     def __init__(self, ptr) -> None:
         self._ptr = ptr
+
+    @classmethod
+    def from_pixels(
+        cls, renderer: Renderer, fmt: int, stride: int, width: int, height: int, data
```

Comment:
What is the `data` here? Is this a `cdata` struct? I'm guessing it's that and not `bytes`.

Reply:
The C signature has `const void *data`. `type()` when I use it gives me `<class '_cffi_backend._CDataBase'>`. cffi has a `CTypesType`, maybe it's that? But yeah it is cdata.

Reply:
OK so turns out they are just `ffi.CData`, which is easy to use.